geode-core/src/main/java/org/apache/geode/cache/client/internal/QueryOp.java

```diff
@@ -120,7 +122,16 @@ protected Object processResponse(Message msg) throws Exception {
           queryResult = resultPart.getObject();
         } catch (Exception e) {
           String s = "While deserializing " + getOpName() + " result";
-          exceptionRef[0] = new SerializationException(s, e);
+          if (e instanceof PdxSerializationException) {
```

Comment:
The conversion of PdxSerializationException to an IOException while reading a query response seems like a hack to fix a problem that we do not yet understand. What do you think of it being disabled by default and only enabled if a system property "gemfire.enableQueryRetryOnPdxSerializationException" is true? This would give the current customer a workaround if they set the new system property. 

You should also document that this workaround will only work if the client is configured to connect to more than one cache server AND the pool's "retry-attempts" is -1 (the default which means try each server) or > 0. It is possible that if you closed the current connection and got a new connection to the same server and retried the query to it that it would also workaround this issue and it would not have the limitations of needing multiple servers and would not depend on the retry-attempts configuration.

Since we have not been able to reproduce this issue can we get the customer to test this workaround in their environment to confirm it will solve their issue?

Reply:
That could be a good idea to introduce the property.

Reply:
Our code will not auto retry the failed server. But I will put into the comments for the property.